client/my-sites/stats/stats-purchase/stats-purchase-checkout-redirect.tsx

```diff
@@ -36,7 +40,7 @@ const getStatsCheckoutURL = (
 	setUrlParam( checkoutProductUrl, 'redirect_to', redirectUrl );
 	setUrlParam( checkoutProductUrl, 'checkoutBackUrl', checkoutBackUrl );
 
-	if ( isFromJetpack && siteSlug ) {
+	if ( useLoggedOutFlow && siteSlug ) {
```

Comment:
I could be wrong but these values are used to redirect user back to WP-Admin correctly, and we don't want to test whether it is the logged out flow and then apply them.

However, this surfaces an issue when `from` is not started with `jetpack`, the flow would end up in Calypso, which we don't want to see. I think we should get the flows fixed first - a possible solution is to prefix anything starting from WP-Admin `jetpack` like the `jetpack-my-jetpack` case? We need another identifier for logged out flow just like what the PR is doing.

Reply:
This maintains the behaviour added by Ian to support the logged out cart. The only difference is that it also applies to sites who come through the `jetpack-stats-upgrade-notice` banner now. All other upgrade paths should stay with the logged in cart.

I don't think updating all our `from` values to start with `jetpack` will fix things here as the logged out version of the cart doesn't handle upgrades correctly. At least not in my understanding of it.

Reply:
Adding jetpack prefix won’t fix the logged out flow, but it’ll fix the returning url for logged in flow if that makes sense. It’s two different things and a couple of places broken here I’m afraid 😟

Reply:
The from here is not the same thing as the from param for the interstitial (the purchase page) if I recall correctly. @dognose24?

Reply:
I was wrong. Please disregard this..